src/getLocale.ts

```diff
@@ -23,6 +24,12 @@ export const __resetCachedValue = (): void => (locale = void 0);
  */
 
 export const getLocale = async (): Promise<CountryCode | null> => {
+	// return locale from cookie if it exists
+	const fromCookie = getCookie('GU_geo_country');
+
+	if (fromCookie) return fromCookie as CountryCode;
```

Comment:
I think we’d want to do a few things here to make this more robust:

1. check whether `isValidCountryCode`
2. move this inside `try / catch`, otherwise we risk having out of sync cookies, memoized variable and local storage.

Reply:
> 1. check whether isValidCountryCode

I added the check although I think this check is really unnecessary since fastly already returns a valid [Two-letter ISO 3166-1 alpha-2 country code for the country. ](https://developer.fastly.com/reference/vcl/variables/geolocation/client-geo-country-code/)


> 2. move this inside try / catch, otherwise we risk having out of sync cookies, memoized variable and local storage.

Why is that? this new cookie will not be cached at all since it will always be returned from fastly on page load time. if this is not the case (e.g localhost or some domain that doesn't go through fastly) the cookie will `null` and then we will continue with that we were doing before - fetch country code from geolocation endpoint and cache it.

Reply:
If we’re memoizing the `locale`, so that it can be accessed without needing an extra check (cookie or local storage), it’s worth leveraging that benefit.

I think that the reading from the cookie should come in place of a `fetch` call to the API, but not as a short-circuit upstream. Otherwise, it would be impossible to force a user into a locale on prod, as the cookie override it every time.

Trying to diagram it, I find `A` chain is easier to reason about than `B`.

```
A. let locale? → localStorage? → cookie? → API!

B. cookie? → let locale? → localStorage? → API!
```

Reply:
Agreed with memoizing it 👍
I've changed it to:
```
C. locale? → cookie? (If yes set to locale)→ localStorage? (If yes set to locale) → API! (Set response to local & localStorage)
```
Does that makes sense?
In prod we shouldn't touch localStorage or API I think.

Reply:
I had a re-think of this and I think keeping localStorage in local could lead to confusion. It would be better to keep the same storage as production - i.e cookie, so thinking of changing to that?

`locale? → cookie? (If yes set to locale)→ API! (Set response to cookie)`